crates/task-impls/src/consensus.rs

```diff
@@ -553,6 +566,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     );
                     consensus.saved_leaves.insert(leaf.commit(), leaf.clone());
 
+                    if let Err(e) = self
+                        .storage
+                        .write()
+                        .await
+                        .update_undecided_state(
```

Comment:
Two questions here:
1. What happens when these start to error constantly? Like, if a write fails or an update, does that signify that the service has degraded?
2. Do we garbage collect old data? For instance if we've moved a few hundred steps, does it make sense to hold onto all the prior state?

Reply:
1. It would have no effect on anything besides being able to recover this state.  It would probably hint at a larger problem with storage but that would also show up when we try to store DA and VID info.
2. Consensus does garbage collect this.  What the storage implementation chooses to do is up for them to decide.  The recommendation would be that every call to `update_X` would rewrite whatever "X" was previously.  In other words, only keep the most recent information